applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -858,6 +946,8 @@ static int discover_source(struct bt_conn *conn)
 	int ret = 0;
 	static uint8_t discover_index;
 
+	memset(temp_codec_cap, 0, sizeof(temp_codec_cap));
```

Comment:
What if we are doing sink discovery for one device and source discovery for the other device at "the same time"?

Reply:
Implemented functionality for handling multiple connections doing discovery at the same time.
We might have to see if there are alternative solutions to storing the capabilities temporarily though. 
It is starting to smell like fish.